### PR TITLE
[keycloakx] Health check depens on health.enabled

### PR DESCRIPTION
**Pull request by Footur**
A potential fix for bug #636.

### CODE REVIEW

File: charts/keycloakx/templates/statefulset.yaml [hunk 1]
```diff
@@ -148,6 +148,7 @@ spec:
             {{- with .Values.extraPorts }}
             {{- toYaml . | nindent 12 }}
             {{- end }}
+          {{- if .Values.health.enabled }}
```
Comment (grieshaber) [inline charts/keycloakx/templates/statefulset.yaml:151]: I'd say we can skip this if clause - if someone decides to stick to the "old" livness-probes, he might not need to enable the health endpoint. But, with the proposal, he needs to enable the endpoint in order to use live-, readiness and startupProbe at all. What do you think?

**Review by grieshaber — CHANGES_REQUESTED** [review]

File: charts/keycloakx/templates/statefulset.yaml (hunk 1 above)
Comment (Footur) [reply charts/keycloakx/templates/statefulset.yaml:151]: I don't see the use case here. Why should anyone want to use the old health checks?

Comment (Footur) [reply charts/keycloakx/templates/statefulset.yaml:151]: The health endpoint is enabled by default. This PR is about using health checks at all. It doesn't matter in this case if the old health checks are configured or not.

Comment (grieshaber) [reply charts/keycloakx/templates/statefulset.yaml:151]: In that matter correct - but there would be a dependency between `health.enabled` and the usage of livenessProbe etc. 
I agree, that the use case for not using health for healthchecks is very abstract and i cannot think of any valid reason.. so let's merge.

**Review by grieshaber — APPROVED** [review]

**Pull request merged**